ProjectLighthouse.Servers.API/Controllers/UserEndpoints.cs

```diff
@@ -50,6 +50,32 @@ public async Task<IActionResult> GetUser(string username)
         return this.Ok(user);
     }
 
+    /// <summary>
+    /// Searches for the user based on the query
+    /// </summary>
+    /// <param name="query">The search query</param>
+    /// <returns>A list of users</returns>
+    /// <response code="200">The list of users, if any were found</response>
+    /// <response code="404">No users matched the query</response>
+    [HttpGet("user_search")]
+    [ProducesResponseType(typeof(User), StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status404NotFound)]
+    public async Task<IActionResult> SearchUsers(string query)
+    {
+        List<User> users = await this.database.Users
+            .Where(u => u.PermissionLevel != PermissionLevel.Banned && u.Username.Contains(query))
+            .Where(u => u.ProfileVisibility == PrivacyType.All) // TODO: change check for when user is logged in
+            .Take(20)
+            .OrderByDescending(b => b.UserId)
```

Comment:
```suggestion
            .OrderByDescending(b => b.UserId)
            .Take(20)
```
The limit command should usually go after the order by but I'm not an SQL expert so take this advice with a grain of salt. EntityFramework might already make this optimization regardless of how you structure the LINQ